### PR TITLE
[Mobile App] Home page animations

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- updates app versions
- removed persisted loading from location error message.
- adds scroll animation to home screen

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

#### Screenshots (optional)
Screenshot_20230316-123534_AirQo

### CODE REVIEW

File: mobile/lib/widgets/custom_widgets.dart
Comment (Baalmart) [inline mobile/lib/widgets/custom_widgets.dart:498]: Could it be possible to just use "child" without any assignment sin ce the value's name is the same as the key's name?

File: mobile/lib/screens/dashboard/dashboard_view_legacy.txt
Comment (Baalmart) [inline mobile/lib/screens/dashboard/dashboard_view_legacy.txt:3]: In the future, could we explore just having this legacy code as actual code with a version number indicated in the file name. That way, we can get to minimise commented out code in our codebase. This applies to other apps as well. Any approach to minimise "commented out" code is welcome.

File: mobile/ios/Runner.xcodeproj/project.pbxproj
Comment (Baalmart) [inline mobile/ios/Runner.xcodeproj/project.pbxproj:461]: Not sure I follow here but I am used to version numbers of anything to always be going higher. 😀😀 please shed more light here?

File: mobile/lib/screens/dashboard/dashboard_view.dart
Comment (NoahNsimbe) [inline mobile/lib/screens/dashboard/dashboard_view.dart:84]: Using silver to add animation to home screen

**Review by Baalmart — COMMENTED** [review]
Thanks @NoahNsimbe for this amazing work! I have left some clarification comments for your review.

Also, I encourage @Mnoble-19 and @BenjaminSsempala to carry out tests (run the app) accordingly. I will then perform merge duty afterwards.

Thanks again!

File: mobile/ios/Runner.xcodeproj/project.pbxproj
Comment (NoahNsimbe) [reply mobile/ios/Runner.xcodeproj/project.pbxproj:461]: This PR was originally meant to update the version codes and create a release. But review took longer than expected. Ended up using it to resolve the error message .

File: mobile/lib/screens/dashboard/dashboard_view_legacy.txt
Comment (NoahNsimbe) [reply mobile/lib/screens/dashboard/dashboard_view_legacy.txt:3]: Thanks @Baalmart. This PR adds animation to the home screen. This file is the version with out animation. It shall be deleted once the UX/UI designer approves the animation otherwise, we shall roll back to this file. The reason it has a different file extension with `// ignore` and `// ignore_for_file` is to prevent dart code metrics from scanning the file and blocking merge operations.

File: mobile/lib/widgets/custom_widgets.dart
Comment (NoahNsimbe) [reply mobile/lib/widgets/custom_widgets.dart:498]: Thanks @Baalmart for this. We can achieve it by using positional parameters. The motivation to use chid is to follow flutter's convention. It uses `child` and `children` named parameters to represent child/children widget(s). 

In this case `AppSafeArea` is a parent widget that `"extends"` flutter's inbuilt  SafeArea widget to include vertical and horizontal padding.

cc @Mnoble-19 @BenjaminSsempala

File: mobile/ios/Runner.xcodeproj/project.pbxproj
Comment (NoahNsimbe) [reply mobile/ios/Runner.xcodeproj/project.pbxproj:461]: This is a build number not a version number.
For iOS, build numbers can be reset for individual version numbers.
For android, build number (version codes) have to be incremented irrespective of the version number.

**Review by BenjaminSsempala — APPROVED** [review]
LGTM

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe

**Pull request merged**
